src/plone/restapi/deserializer/utils.py

```diff
@@ -25,11 +25,13 @@ def path2uid(context, link):
         )
 
     # handle edge-case when we have non traversable path like /@@download/file
-    if "/@@" in path:
-        path, suffix = path.split("/@@", 1)
-        suffix = "/@@" + suffix
-    else:
-        suffix = ""
+    SUFFIXES = ["/@@", "#"]
+    suffix = ""
+    for suffix_separator in SUFFIXES:
+        if suffix_separator in path:
+            path, suffix = path.split(suffix_separator, 1)
```

Comment:
@sneridagh You are overwriting `suffix` here. So if both separators are present, only one of the suffixes will be kept.

Reply:
It seemed wrong, but it's fine, I added a test. Ported the pattern to the serializer too.